### ISSUE TITLE
Assertion failure when trying to read XML Schema

### ISSUE THREAD

**Issue by smandell** [peerlibrary/node-xml4js#14]
I've getting an assertion failed error when trying to load and use a schema file that I'm providing. I've gotten as far as I can stepping through the code, and don't understand what exactly the issue is being caused by. 

I've attached the schema file that I'm passing into .addSchema(). You just need to rename the attached file to .xsd. 

[schema.txt](https://github.com/peerlibrary/node-xml4js/files/681561/schema.txt)

**Comment by mitar** [comment]
Sadly not complete XML schema is supported yet. So it might be that you are finding something like that.

**Comment by smandell** [comment]
@mitar - did you determine that the schema was not complete from looking at it? This is a schema that is provided by a software company of their product, so if it's not complete, or has an issue, I can file a bug against them, but I could use your help in understanding what about it isn't complete. 

I ran it though a XML validator and it appears to be valid XML.

**Comment by mitar** [comment]
No, what I am saying is that this library does not support the full range of XML schema. So your schema probably uses constructs or directives not supported by this library.

**Comment by smandell** [comment]
Understood, thanks for clarifying. Should you ever feel the itch to add additional support to the tool I'd be grateful. I tried to debug on my own but am hitting the limitations of my current abilities.

**Comment by mitar** [comment]
Sadly, at the moment I do not have time. :-(
